bokehjs/src/coffee/api/charts.coffee

```diff
@@ -0,0 +1,138 @@
+_ = require("underscore")
+$ = require("jquery")
+sprintf = require("sprintf")
+{Document} = require("../document")
+embed = require("../embed")
+models = require("./models")
+palettes = require("../palettes/palettes")
+
+sum = (array) ->
+  return array.reduce(((a, b) => a + b), 0)
+
+cumsum = (array) ->
+  result = []
+  array.reduce(((a, b, i) -> result[i] = a + b), 0)
+  return result
+
+num2hexcolor = (num) -> sprintf("#%06x", num)
+
+pie = (data, opts) ->
```

Comment:
FYI Chris we currently have zero experience documenting JS libraries. If you have suggestions for best or common tools or formats, especially that could integrate well into our existing docs site, they would be appreciated.